### PR TITLE
feat: [SSCA-288] add resources config to SSCA CI steps, fix empty option coming in dropdown artifact stage selector

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary
cherry pick changes from 
https://github.com/harness/harness-core-ui/pull/16758
https://github.com/harness/harness-core-ui/pull/16808

image
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepEdit.tsx [hunk 1]
```diff
@@ -367,19 +355,77 @@ const SscaEnforcementStepEdit = (
...
+                              expressions,
+                              allowableTypes,
+                              disabled: readonly
+                            }}
+                            tooltipProps={{ dataTooltipId: 'setContainerResources' }}
+                          />
+                          {getMultiTypeFromValue(get(formik.values, 'spec.resources.limits.cpu')) ===
+                            MultiTypeInputType.RUNTIME &&
+                            !readonly && (
+                              <ConfigureOptions
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepEdit.tsx:413]: Are we not supporting allowed values here?

File: src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx [hunk 2]
```diff
@@ -17,56 +18,53 @@ import {
...
   return input.map(item => ({ label: item, value: item }))
 }
 
 const artifactTypeOptions = getTypedOptions<SbomSource['type']>(['image'])
 const sbomGenerationToolOptions = getTypedOptions<SbomOrchestrationTool['type']>(['Syft'])
-const syftSbomFormats: { label: string; value: string }[] = [
+const syftSbomFormats: { label: string; value: SyftSbomOrchestration['format'] }[] = [
   { label: 'SPDX', value: 'spdx-json' },
-  { label: 'CYCLONEDX', value: 'cyclonedx-json' }
+  { label: 'CycloneDX', value: 'cyclonedx-json' }
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx:64]: These labels should come from yaml

File: src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx [hunk 3]
```diff
@@ -308,11 +297,11 @@ const SscaOrchestrationStepEdit = (
                       placeholder={getString('pipeline.infraSpecifications.namespacePlaceholder')}
                       multiTextInputProps={{ expressions, textProps: { disabled: readonly }, allowableTypes }}
                     />
-                    {getMultiTypeFromValue(formik.values.spec.infrastructure?.spec.namespace) ===
+                    {getMultiTypeFromValue(get(formik.values, 'spec.infrastructure.spec.namespace')) ===
                       MultiTypeInputType.RUNTIME &&
                       !readonly && (
                         <ConfigureOptions
-                          value={formik.values.spec.infrastructure?.spec.namespace as string}
+                          value={get(formik.values, 'spec.infrastructure.spec.namespace')}
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx:304]: Is allowed values support not needed here?

File: src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx (hunk 2 above)
Comment (udayvunnam) [reply src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx:64]: Agreed, will get towards it.

File: src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx (hunk 3 above)
Comment (udayvunnam) [reply src/modules/75-ssca/components/PipelineSteps/common/SscaOrchestrationStepEdit.tsx:304]: This was an existing component. Will check for this in the latest develop

**Pull request merged**